Repo reorg - Break out Nextjs, Flask and Streamlit

lopesa opened the pull request:
This is a large reorganization of the repo now that keeps open our two options of a final product and creates a central workspace folder. The folder structure looks like this now:

/
| - /**backend_flask** - (backend of 2 pc solution)
| - /**ui_next** - (frontend of 2 pc solution)
| - / **streamlit_app** - (standalone app)
| - /**python_workspace** - (general workspace)
| - /**pdf** - (source materials for mvp)

On commit to the main branch, all pieces deploy automatically, to the following hosts:
backend_flask --> Digital Ocean App Platform
ui_next --> vercel
streamlit_app --> streamlit

as of this commit everything is working except for the streamlit app's connections with the vectorstore which I plan to look at next. (The Next/Flask app doesn't try to use the vectorstore). For Streamlit specifically, Chroma seems integrated so I should be able to follow this.

One notable weird thing with the repo after the reorg is that to start the Flask/Next app the dev has to go into the next_ui folder and run npm commands. Now. it seems like it would be great if we can run Poetry commands from the root folder but
I didn't spend time messing with it. I can also look at that next, unless anyone sees a clear solution

lopesa committed:
moves the api into a /backend folder and adds gunicorn to serve on app server

lopesa committed:
replace poetry.lock file

lopesa committed:
fix flask-dev-monorepo task in npm and restore /api/chat from hello world to chat

lopesa committed:
big commit moving the nextjs ui app into a ui folder

lopesa committed:
update next.config to redirect /api routes in prod to DO app

- not vercel edge functions

lopesa committed:
create a streamlit_app folder for deploy and put COPIES

- of relevant files in folder, temporarily copies

lopesa committed:
generate a new requirements.txt file and copy out to /backend and /streamlit_app

- triggers the formatting pre-commit hook, so there's formatting changes

lopesa committed:
add pysqlite3 on deployed version to satisfy Chroma requirements

- add researched from internet. Also have to comment this out
- on mac silicon, (find the Python env variable for machine type)

lopesa committed:
final names for folders and update readme

curious-llama reviewed (approved):
This looks great!

curious-llama commented:
```
to start the Flask/Next app the dev has to go into the next_ui folder and run npm commands. Now. it seems like it would be great if we can run Poetry commands from the root folder
```

Try running this from root: `poetry run python -m flask --app backend_flask.app run -p 5328` in order to start the backend. Looks like that works for me. 

We can add a Makefile if we want have an alias for this, like we used to have with the next.js 'flask-dev'.

lopesa committed:
resolve merge conflicts, and

- update package json to not include any Flask app control
- update README to include new poetry based commands for Flask and Streamlit apps
- lots of prettier made formatting changes in here
- i did confirm app were working locally before pushing, not deployed though
  will address in future commit

lopesa commented:
@coding-connor thanks! that poetry command also works for me and gets us away from using package.json to start the python app, lol. I added it to the readme and also removed the command to start flask from package.json
